chemprop/nn/loss.py

```diff
@@ -348,3 +349,30 @@ def _calc_unreduced_loss(self, preds: Tensor, targets: Tensor, mask: Tensor, *ar
 
     def extra_repr(self) -> str:
         return f"threshold={self.threshold}"
+
+
+@LossFunctionRegistry.register(["quantile", "pinball"])
+class QuantileLoss(LossFunction):
+    def __init__(self, task_weights: ArrayLike = 1.0, alpha: float = 0.1):
+        super().__init__(task_weights)
+        self.alpha = alpha
+
+    def _calc_unreduced_loss(self, preds: Tensor, targets: Tensor, mask: Tensor, *args) -> Tensor:
+        mean, interval = torch.unbind(preds, dim=-1)
+
+        device = interval.device
+        self.bounds = torch.tensor([-1 / 2, 1 / 2], device=device).view(2, 1, 1)
+        self.tau = torch.tensor(
+            [[self.alpha / 2, 1 - self.alpha / 2], [self.alpha / 2 - 1, -self.alpha / 2]],
+            device=device,
+        ).view(2, 2, 1, 1)
```

Comment:
if `alpha` is only used insofar as it defines `self.tau`, why are we recalculating it in _every_ call to `_calc_unreduced_loss()`? This function is called **every batch**, so we're repeating the same work $b$ times every epoch. I don't think it matters inasmuch as this work is minimal, but I think it actually harms readability IMO.

Reply:
Originally, I registered `bound` and `tau` as buffers so they would be saved in the state dict. However, since they can be recreated from `alpha`, I removed their registration as buffers. Do you have any suggestions for this? Maybe add an if statement in `_calc_unreduced_loss` to check if the devices are consistent. If not, move them to the correct device?

Reply:
What's wrong with registering them as buffers? I don't see a problem with storing these as buffers and then _also_ storing `alpha` even though it's not actually used and thus "wasting" memory (but it's really only 8 bytes of extra storage _per object_, and we'll likely only have 1 object alive at any time, so the impact is negligible.